platform/ios/src/MGLMapboxEvents.m

```diff
@@ -433,51 +434,63 @@ - (MGLMapboxEventAttributes *)fullyFormedEventForEvent:(NSString *)event withAtt
     return nil;
 }
 
+- (void)updateDictionary:(NSMutableDictionary *)dictionary withKey:(NSString *)key value:(id)value {
+    if (key && value) {
+        [dictionary setValue:value forKey:key];
+    }
+}
+
 - (MGLMapboxEventAttributes *)locationEventWithAttributes:(MGLMapboxEventAttributes *)attributeDictionary {
-    MGLMutableMapboxEventAttributes *attributes = [@{MGLEventKeyEvent: MGLEventTypeLocation,
-                                                     MGLEventKeySource: MGLEventSource,
-                                                     MGLEventKeySessionId: self.instanceID,
-                                                     MGLEventKeyOperatingSystem: self.data.iOSVersion} mutableCopy];
+    MGLMutableMapboxEventAttributes *attributes = [NSMutableDictionary dictionary];
+    [self updateDictionary:attributes withKey:MGLEventKeyEvent value:MGLEventTypeLocation];
+    [self updateDictionary:attributes withKey:MGLEventKeySource value:MGLEventSource];
+    [self updateDictionary:attributes withKey:MGLEventKeySessionId value:self.instanceID];
```

Comment:
@1ec5 it would be, but I'd like to run everything through the `updateDictionary:...` method (even if it is not always necessary) for consistency. Some of the values could be nil (either as a matter of course or because of bugs). The utility method helps ensure that we don't crash because of such a mistake. Always using the utility method instead of a mix of the utility method and `dictionary[key] = value` is easier to read, I think.